akka-actor/src/main/scala/akka/io/dns/CachePolicy.scala

```diff
@@ -15,7 +15,7 @@ object CachePolicy {
   case object Forever extends CachePolicy
 
   final class Ttl private (val value: FiniteDuration) extends CachePolicy {
-    if (value <= Duration.Zero) throw new IllegalArgumentException(s"TTL values must be a positive value.")
+    if (value < Duration.Zero) throw new IllegalArgumentException(s"TTL values must be a positive value.")
```

Comment:
I think we should add the more strict check in `fromPositive` and add a separate factory method for the zero Ttl.

```
    def fromPositive(value: FiniteDuration): Ttl = {
      if (value <= Duration.Zero) throw new IllegalArgumentException(s"TTL values must be a positive value.")
      new Ttl(value)
    }

    def never: Ttl = new Ttl(Duration.Zero)
```

Do we also have to change somewhere else where `fromPositive` is used, such as in `ResourceRecord.parse`? There it could use `new Ttl` instead of the value can be 0.

Reply:
@patriknw Yes, sorry I was getting around to that. I was just pondering about the mathematical meaning of "positive values". Depending on the context, zero is neither positive nor negative, but in the set of real numbers there's a subset called strictly positive real numbers. [Source](https://proofwiki.org/wiki/Definition:Strictly_Positive). 

I'll push a new commit shortly.